libs/wire-api/src/Wire/API/User/Client/DPoPAccessToken.hs

```diff
@@ -0,0 +1,127 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+-- This file is part of the Wire Server implementation.
+--
+-- Copyright (C) 2022 Wire Swiss GmbH <opensource@wire.com>
+--
+-- This program is free software: you can redistribute it and/or modify it under
+-- the terms of the GNU Affero General Public License as published by the Free
+-- Software Foundation, either version 3 of the License, or (at your option) any
+-- later version.
+--
+-- This program is distributed in the hope that it will be useful, but WITHOUT
+-- ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+-- details.
+--
+-- You should have received a copy of the GNU Affero General Public License along
+-- with this program. If not, see <https://www.gnu.org/licenses/>.
+
+module Wire.API.User.Client.DPoPAccessToken where
```

Comment:
jwt-tools should provide a high-level, but generic interface to the rust code.  i think your goal was to make the interface as low-level as possible

example: `DPoPTokenGenerationError` isn't strictly needed to import the rust code, but it makes using the rust code more convenient without loss of generality.

also, don't import wire-api in jwt-tools.  :-)

Reply:
on the other hand, we probably don't want to talk about the servant api inside jwt-tools, and that may account for almost everything in this module.  or you'd have to wrap everything in newtypes to avoid orphans.